src/particles/particles.cpp

```diff
@@ -0,0 +1,55 @@
+#include "particles.h"
+
+#include <algorithm>
+#include <cctype>
+#include <stdexcept>
+
+Particle::Particle(long n) : n(n) {
```

Comment:
Rename `n` to `id`, to serve such role as this: https://github.com/libamtrack/pyamtrack/blob/083156389421d2e91b71a1529bf0be82d14bc638/src/materials/materials_bindings.cpp#L19